core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -356,11 +359,34 @@ private void setFinalResult(
               TimeUnit.NANOSECONDS);
         }
       }
+      // log the warnings if they have NOT been disabled
+      if (LOG.isWarnEnabled()
+          && !executionInfo.getWarnings().isEmpty()
+          && !executionProfile.getBoolean(DefaultDriverOption.REQUEST_DISABLE_LOG_WARNINGS)) {
+        logServerWarnings(executionInfo.getWarnings());
+      }
```

Comment:
Note: the `if (result.complete(resultSet)) { ... }` block above is in case another speculative execution beat us to completing the result.
We log the warnings outside of that block, which means we'll do so even if we raced. I think I'm OK with it but I thought I'd just mention it in passing.